File: delite/generatedCache/cpp/src/kernels/x609.cpp
//activation record for fat loop
#ifndef __activation_x609__
#define __activation_x609__
#include "cpphelperFuncs.h"
class activation_x609 {
public:
activation_x609 *left_act;
cppDeliteArraydouble *x609;
cppDeliteArraydouble *x609_data;
};
#endif
#include "cpphelperFuncs.h"
#ifndef __kernel_x609__
#define __kernel_x609__
class kernel_x609{
public:
int32_t x557;
cppDeliteArraydouble *x558;
cppDeliteArraydouble *x559;

kernel_x609(int32_t  _x557,cppDeliteArraydouble * _x558,cppDeliteArraydouble * _x559) {
x557 = _x557;
x558 = _x558;
x559 = _x559;
}

int32_t size(resourceInfo_t &resourceInfo) {
return x557;}

int32_t loopStart;
int32_t loopSize;
activation_x609 *alloc(resourceInfo_t &resourceInfo) {
activation_x609 * __act = new activation_x609();;
int32_t x571 = loopSize;
cppDeliteArraydouble *x575 = new (resourceInfo.thread_id) cppDeliteArraydouble(x571, resourceInfo.thread_id);
__act->x609_data = x575;
return __act;}

activation_x609 *processRange(resourceInfo_t &resourceInfo,activation_x609 *__act,int32_t start,int32_t end) {
bool  isEmpty = end-start <= 0;
activation_x609 * __act2 = init(resourceInfo,__act,start,isEmpty);
int32_t  idx = start + 1;
while (idx < end) {
process(resourceInfo,__act2,idx);
idx = idx + 1;
}
return __act2;}

activation_x609 *init(resourceInfo_t &resourceInfo,activation_x609 *__act,int32_t x242,bool isEmpty) {
if (!isEmpty) {
process(resourceInfo,__act,x242);
}
return __act;}

void process(resourceInfo_t &resourceInfo,activation_x609 *__act,int32_t x242) {
double x566 = x558->apply(x242);
double x567 = x559->apply(x242);
double  x568 = x566 * x567;
double x570 = x568;
cppDeliteArraydouble *x569 = __act->x609_data;
x569->update(x242, x570);
}

void combine(resourceInfo_t &resourceInfo,activation_x609 *__act,activation_x609 *rhs) {
}

void postCombine(resourceInfo_t &resourceInfo,activation_x609 *__act,activation_x609 *lhs) {
__act->left_act = lhs;
}

void postProcInit(resourceInfo_t &resourceInfo,activation_x609 *__act) {
}

void postProcess(resourceInfo_t &resourceInfo,activation_x609 *__act) {
}

void finalize(resourceInfo_t &resourceInfo,activation_x609 *__act) {
cppDeliteArraydouble * x569 = __act->x609_data;
__act->x609 = x569;
}

activation_x609 *initAct(resourceInfo_t &resourceInfo) {
activation_x609 * act = new activation_x609;
return act;}

};
#endif

File: delite/generatedCache/scala/src/kernels/x609.scala
package generated.scala
//activation record for fat loop
final class activation_x609 {
var left_act: activation_x609 = _
var x609: Array[Double] = _
var x609_data: Array[Double] = _
}
object activation_x609 {
}
object kernel_x609 {
def apply(x557:Int,x558:Array[Double],x559:Array[Double]): generated.scala.DeliteOpMultiLoop[activation_x609] = {
val x609 = new generated.scala.DeliteOpMultiLoop[activation_x609] {
def size(): Int = {
x557}

var loopStart: Int = _
var loopSize: Int = _
def alloc(): activation_x609 = {
val __act: activation_x609 = new activation_x609()
val x571 = loopSize
val x575 = new Array[Double](x571)
__act.x609_data = x575
__act}

def processRange(__act:activation_x609,start:Int,end:Int): activation_x609 = {
val isEmpty: Boolean = end-start <= 0
val __act2: activation_x609 = init(__act,start,isEmpty)
var idx: Int = start + 1
while (idx < end) {
process(__act2,idx)
idx = idx + 1
}
__act2}

def init(__act:activation_x609,x242:Int,isEmpty:Boolean): activation_x609 = {
if (!isEmpty) {
process(__act,x242)
}
__act}

def process(__act:activation_x609,x242:Int): Unit = {
val x566 = x558(x242)
val x567 = x559(x242)
val x568 = {
x566 * x567}
val x570 = x568
val x569 = __act.x609_data
val x579 = x569(x242) = x570
}

def combine(__act:activation_x609,rhs:activation_x609): Unit = {
}

def postCombine(__act:activation_x609,lhs:activation_x609): Unit = {
__act.left_act = lhs
}

def postProcInit(__act:activation_x609): Unit = {
}

def postProcess(__act:activation_x609): Unit = {
}

def finalize(__act:activation_x609): Unit = {
var x569: Array[Double] = __act.x609_data
__act.x609_data = null
__act.x609 = x569
}

def initAct(): activation_x609 = {
val act: activation_x609 = new activation_x609
act}

}

x609
}}
